iam.tf

```diff
@@ -168,6 +168,42 @@ resource "aws_iam_policy_attachment" "sqs" {
   policy_arn = aws_iam_policy.sqs[0].arn
 }
 
+#############
+# SNS Config
+#############
+
+data "aws_iam_policy_document" "sns" {
+  count = local.create_role && var.attach_sns_policy ? 1 : 0
+
+  statement {
+    sid    = "SNSAccess"
+    effect = "Allow"
+    actions = [
```

Comment:
Split this policy statement into 2 - one for SNS and one for KMS actions. They should have different `resources`.

Reply:
What do you mean they should have different `resources`. Should the second statement which allows KMS actions not target the SNS resource. I added the KMS actions as an afterthought to published encrypted messages. https://repost.aws/knowledge-center/sns-not-getting-eventbridge-notification

I can see the existing SQS example uses a single policy statement.

Reply:
`resources = ???` => `resources = ["*"]` as it says in "Example IAM policy statement that allows EventBridge to publish messages to an encrypted Amazon SNS topic" section on https://repost.aws/knowledge-center/sns-not-getting-eventbridge-notification

Reply:
Oh i see, good spot